dao/src/main/scala/za/co/absa/enceladus/dao/rest/RestClient.scala

```diff
@@ -76,23 +83,37 @@ protected class RestClient(authClient: AuthClient,
 
     val statusCode = response.getStatusCode
 
+    /**
+     * This function handles unauthorized response by trying to authenticate
+     * (if there are still some retries attempt left) - this might be due to an expired session.
+     *
+     * @param exceptionToThrow Instance of an exception that should be thrown if there are no retries left.
+     */
+    def handleUnauthorizedResponse(exceptionToThrow: OptionallyRetryableException): Unit = {
```

Comment:
```suggestion
    def handleUnauthorizedResponse(exceptionToThrow: =>OptionallyRetryableException): Unit = {
```
In this situation it's nice to use _by-name parameter_
Explanation can be found here for example:
https://www.tutorialspoint.com/scala/functions_call_by_name.htm
https://alvinalexander.com/scala/fp-book/how-to-use-by-name-parameters-scala-functions/